### PR TITLE
feat: set gpo task

### PR DESCRIPTION
**Pull request by karlfloersch**
**Description**
Add a task for setting the gas price oracle.

[from the commit message, contributed by @tynes]

Creates a task that can set the owner of any contract that is `Ownable`.
It defaults to attempting to set the address of the `OVM_GasPriceOracle`
on L2. The following config options must be set:

- `CONTRACTS_RPC_URL`
- `CONTRACTS_DEPLOYER_KEY`

It can be invoked like so:

```
$ npx hardhat set-owner --owner 0x..
```

On L2 networks that do not enforce transaction fees, the
`--transaction-gas-price 0` config option may be used for
convenience.

This script also supports Ledger with the `--use-ledger` flag.
It defaults to deriving a key at the `ethers.utils.defaultPath`.

### CODE REVIEW

**Comment by tynes** [comment]
This could technically be `set-owner` and work with any `Ownable`

**Review by tynes — APPROVED** [review]

File: packages/contracts/tasks/set-owner.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import { LedgerSigner } from '@ethersproject/hardware-wallets'
+import { task } from 'hardhat/config'
+import * as types from 'hardhat/internal/core/params/argumentTypes'
+
+import { predeploys } from '../src/predeploys'
+import { getContractFactory } from '../src/contract-defs'
+
+task('set-owner')
+  .addParam('owner', 'the new oracle address', 0, types.string)
+  .addOptionalParam('transactionGasPrice', 'tx.gasPrice', undefined, types.int)
```
Comment (K-Ho) [inline packages/contracts/tasks/set-owner.ts:12]: if this is intended for only L2 we might want to get rid of this optional param

File: packages/contracts/tasks/set-owner.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    if (owner !== addr) {
+      throw new Error(`Incorrect key. Owner ${owner}, Signer ${addr}`)
+    }
+
+    console.log(`Owner is currently ${owner.toString()}`)
+    console.log(`Setting owner to ${args.owner}`)
+
+    const tx = await Ownable.connect(signer).transferOwnership(
+      args.owner,
+      { gasPrice: args.transactionGasPrice }
```
Comment (K-Ho) [inline packages/contracts/tasks/set-owner.ts:68]: Follow up to above comment to remove this if this is intended for L2

File: packages/contracts/tasks/set-owner.ts (hunk 1 above)
Comment (tynes) [reply packages/contracts/tasks/set-owner.ts:12]: We want to be able to specify `gasPrice: 0` on L2 for simplicity of usage, see the commit message for rationale

**Pull request merged**